test/LRPTests/APITests.cpp

```diff
@@ -30,7 +30,7 @@ namespace Test::LRP
 
         winrt::com_ptr<INotificationsLongRunningPlatform> GetNotificationPlatform()
         {
-            winrt::com_ptr notificationPlatform{ winrt::try_create_instance<INotificationsLongRunningPlatform>(_uuidof(NotificationsLongRunningPlatform), CLSCTX_ALL) };
+            winrt::com_ptr notificationPlatform{ winrt::try_create_instance<INotificationsLongRunningPlatform>(CLSID_NotificationsLongRunningPlatform, (CLSCTX_LOCAL_SERVER | CLSCTX_ALLOW_LOWER_TRUST_REGISTRATION)) };
```

Comment:
Why remove CLSCTX_ALL? It should cover all flags including local server and lower trust right?

Reply:
CLSCTX_ALL isn't all of the CLSCTX flags:
`#define CLSCTX_ALL              (CLSCTX_INPROC_SERVER| \
                                 CLSCTX_INPROC_HANDLER| \
                                 CLSCTX_LOCAL_SERVER| \
                                 CLSCTX_REMOTE_SERVER)`

CLSCTX_ALL is for "I have a CLSID and I want to activate it, but I don't know/care whether it's inproc, out-of-proc, on a completely different machine, etc.".
In theory, that's a reasonable default perspective: the client is agnostic about where the server class actually runs, allowing the server location to be treated as an implementation detail. In practice, the fact that CoCreateInstance _allows_ the client to specify this prevents any such agnosticism from being useful. If I want to change my server's activation type from out-of-proc to inproc, I can't unless I'm willing to break any existing clients which pass CLSCTX_LOCAL_SERVER instead of CLSCTX_ALL.

So CLSCTX_ALL doesn't end up adding much real value in most scenarios. In cases like this where your client and server are tightly coupled and the client knows the server is out-of-proc, you might as well just pass CLSCTX_LOCAL_SERVER. It makes the relationship to the server a little clearer, and should also have very minor perf benefits (e.g. avoids some registry queries).

Reply:
Thanks for the detailed explanation. Its good to understand some of the intricacies of COM that we simply take for granted when coding solutions up. I incorrectly assumed ALL mapped to everything. But in any case, it sounds like we shouldn't be using the Big Hammer(Perf, Tight coupling) and it is important to be specific when it comes to COM flags.